addon-test-support/scroll-into-view-mock.js

```diff
@@ -0,0 +1,31 @@
+import sinon from 'sinon';
+import { find } from '@ember/test-helpers';
+export default function setupScrollIntoView(hooks) {
+  hooks.beforeEach(function () {
+    this.scrollIntoViewSandbox = sinon.createSandbox();
+    const scrollIntoViewMock = this.scrollIntoViewSandbox.stub();
+    this.scrollIntoViewSandbox.replace(
+      window.Element.prototype,
+      'scrollIntoView',
+      scrollIntoViewMock
+    );
+    // function will check if the passed selector matches any of the elements
+    this.scrollIntoViewCalledWith = (selector) => {
```

Comment:
@ygongdev @jdkahn I'm curious to get your thoughts here. Should we put the assert method on `this` or return it from the setup method?

This approach is nice as-written, because it means that invoking it (as seen on line 10 of the test below) feels like other setup functions, but it's less declarative because it becomes a magic method that you just have to know to call by reading the documentation. So, I'm a bit torn.

Reply:
@elwayman02 I'm not sure I understand the alternative you're presenting. Are you saying that we have this as a setup function that is used like this
```
hooks.beforeEach(function() {
    this.scrollIntoViewCalledWith = setupScrollIntoView();
}

hooks.afterEach(function() {
    teardownScrollIntoView();
}

test('x', async function(assert) {
    await render(...);
    assert.ok(this.scrollIntoViewCalledWith(...));
}
```